### ISSUE TITLE
Add ability to base on image stream tags

### ISSUE THREAD

**Issue by wtrocki** [aerogearcatalog/data-sync-apb#22]
###  Idea

Currently when installing sync from APB users will have hard time to update the server without manual changes. When using some specific tag like latest etc. image is not going to be fetched when it's already in the OpenShift registry. Standard aproach for this problem is to use OpenShift image streams.
If we introduce image streams we will be able to allow users to try and test this changes. 
We can have separate streams for release and alpha. 
This aproach will be also useful to update current infrastructure that is being hosted on community cluster.

###  How

Add image stream object and point it to latest (tags will be pointed to release tags).
Use rolling deployment strategy based on image stream.

ping for opinions @darahayes @psturc @jhellar 

ping @johnfriz for visibility. 
I believe that this is something that all our APB could have in long run.

**Comment by laurafitzgerald** [comment]
@wtrocki I haven't worked with apbs so this could be a silly question but would `ImagePullPolicy: Always` solve this issue?

**Comment by laurafitzgerald** [comment]
also @wtrocki if this work is required for us as part of development/release work or for the user to remove manual steps can you create a JIRA and bring it into the sprint.

**Comment by wtrocki** [comment]
Yes. `ImagePullPolicy: Always` may be very good idea if we going to label our images with 1.0.0-latest or something like this! This is more about how we will aproach the updates for end users.

**Comment by psturc** [comment]
This is what we need: https://docs.openshift.com/container-platform/3.10/dev_guide/managing_images.html#importing-tag-and-image-metadata
By default an image repository is checked for new image every 15 minutes and multiple image tags can be specified.
I did a little experiment with this on OCP and OpenShift Online (OO) and it seems that OO does not support this feature (that image pull policy is specified at a cluster level and customer can do nothing about it when using OO).
It could be also worth checking this functionality on OSD.

If you agree, I would create a ticket and bring it to the current sprint.

**Comment by wtrocki** [comment]
@psturc Awesome work! Thank you for checking this on many clusters.

I think this will be really tied with versioning discussion that is currently in the progress. There will be couple ways to do it. When I initially proposed that change I wasn't aware of the custom OpenShift UI. We will need to get consensus on this topic and stackholders approval: https://groups.google.com/forum/#!topic/aerogear/WchCNefaQGU
Maybe we can post this on this topic to provide more information.

From my point of view it's is ok to add spike ticket in Jira, but we are still not clear on this land.
